File: pattern5/adapterdesignpattern.cpp
#include<iostream>
using  namespace std;

//Adaptee
class LegacyCode{
    public:
    void getCustomerDataInfo(string id){
      cout<<"customer data fetched for id"<<id<<endl;
    }
};
//target interface or new system
class interface{
    public:
     virtual void getCustomerData(string id, string name)=0;
};
//Adapter
class LegacyCodeAdapter : public interface{
    LegacyCode*legacyCode;
    public:
    LegacyCodeAdapter( LegacyCode* legacyCode){
        this->legacyCode=legacyCode;
    }
    void getCustomerData(string id, string name){
        legacyCode->getCustomerDataInfo(id);
    }
     
};

int main(){

    LegacyCodeAdapter *legacyAdapter= new LegacyCodeAdapter(new LegacyCode());
    legacyAdapter->getCustomerData("123","ankit");

    return 0;
}

File: pattern5/AdapterDesignPattern.java
package pattern5;
//Adaptee
class LegacyCode{
    public void getCustomerDataInfo(String id){
        System.out.println("customer data fetched for id "+id);
    }
}

//target interface
interface CustomerDataFetcher{
     void getCustomerData(String id,String name);
}


//Adapter
class LegacyCodeAdapter implements CustomerDataFetcher{
    LegacyCode legacyCode;
    LegacyCodeAdapter(LegacyCode legacy){
         this.legacyCode=legacy;
    }
    public void getCustomerData(String id, String name){
        legacyCode.getCustomerDataInfo(id);
    }
}

public class AdapterDesignPattern {
    public static void main(String []args){
        LegacyCodeAdapter l1=new LegacyCodeAdapter(new LegacyCode());
        l1.getCustomerData("123", "tata");
    }
    
}
